core/nuxeo-document.js

```diff
@@ -226,30 +226,58 @@ import './nuxeo-resource.js';
       return path;
     }
 
-    appendClientReason(prop) {
-      prop.viewData = `${prop.data}&clientReason=view`;
-      prop.downloadData = `${prop.data}&clientReason=download`;
+    appendClientReason(documentProp) {
+      let sourceUrl = '';
+      if (documentProp) {
+        if ('url' in documentProp) {
+          sourceUrl = 'url';
+        } else if ('data' in documentProp) {
+          sourceUrl = 'data';
+        }
+      }
+      if (sourceUrl) {
+        documentProp.viewUrl = `${documentProp[sourceUrl]}${
+          documentProp[sourceUrl].indexOf('?') > -1 ? '&clientReason=view' : '?clientReason=view'
+        }`;
+        documentProp.downloadUrl = `${documentProp[sourceUrl]}${
+          documentProp[sourceUrl].indexOf('?') > -1 ? '&clientReason=download' : '?clientReason=download'
```

Comment:
Kindly remove repetitive code. Optimize it as mentioned below.

const splitter = documentProp[sourceUrl].indexOf('?') > -1 ? '&' : '?'
documentProp.viewUrl = `${documentProp[sourceUrl]}${splitter}clientReason=view`
documentProp. downloadUrl = `${documentProp[sourceUrl]}${splitter}clientReason= download `